pkg/scheduler/framework/plugins/podtopologyspread/scoring_test.go

```diff
@@ -506,30 +510,91 @@ func BenchmarkTestCalculateEvenPodsSpreadPriority(b *testing.B) {
 	for _, tt := range tests {
 		b.Run(tt.name, func(b *testing.B) {
 			existingPods, allNodes, filteredNodes := st.MakeNodesAndPodsForEvenPodsSpread(tt.pod.Labels, tt.existingPodsNum, tt.allNodesNum, tt.filteredNodesNum)
+			state := framework.NewCycleState()
 			snapshot := nodeinfosnapshot.NewSnapshot(nodeinfosnapshot.CreateNodeInfoMap(existingPods, allNodes))
-			tpSpreadMap, err := buildPodTopologySpreadMap(tt.pod, filteredNodes, snapshot.NodeInfoList)
-			if err != nil {
-				b.Fatal(err)
-			}
-			meta := &priorityMetadata{
-				podTopologySpreadMap: tpSpreadMap,
+			p := &PodTopologySpread{sharedLister: snapshot}
+
+			status := p.PostFilter(context.Background(), state, tt.pod, filteredNodes, nil)
+			if !status.IsSuccess() {
+				b.Fatalf("unexpected error: %v", status)
 			}
 			b.ResetTimer()
 
 			for i := 0; i < b.N; i++ {
 				var gotList framework.NodeScoreList
 				for _, n := range filteredNodes {
 					nodeName := n.Name
-					nodeScore, _ := CalculateEvenPodsSpreadPriorityMap(tt.pod, meta, snapshot.NodeInfoMap[nodeName])
-					gotList = append(gotList, nodeScore)
+					score, status := p.Score(context.Background(), state, tt.pod, nodeName)
+					if !status.IsSuccess() {
+						b.Fatalf("unexpected error: %v", status)
+					}
+					gotList = append(gotList, framework.NodeScore{Name: nodeName, Score: score})
 				}
-				err := CalculateEvenPodsSpreadPriorityReduce(tt.pod, meta, snapshot, gotList)
-				if err != nil {
-					b.Fatal(err)
+
+				status = p.NormalizeScore(context.Background(), state, tt.pod, gotList)
+				if !status.IsSuccess() {
+					b.Fatal(status)
 				}
 			}
 		})
 	}
 }
 
-var softSpread = v1.ScheduleAnyway
+// The tests in this file compare the performance of SelectorSpreadPriority
+// against EvenPodsSpreadPriority with a similar rule.
+
+var (
+	tests = []struct {
```

Comment:
does this really need to be outside the test function?

Reply:
Technically no. I just moved them as they were - from `priorities/spreading_perf_test.go`, as the intention was to use the same test cases to compare the perf of pod topology spread and default selector spreading. I believe once @alculquicondor done the migration of selector spreading, we can find a better place to hold it.